hledger-lib/Hledger/Data/Types.hs

```diff
@@ -230,7 +230,8 @@ data Amount = Amount {
       aprice      :: !(Maybe AmountPrice)  -- ^ the (fixed, transaction-specific) price for this amount, if any
     } deriving (Eq,Ord,Generic,Show)
 
-newtype MixedAmount = Mixed [Amount] deriving (Eq,Ord,Generic,Show)
+newtype MixedAmount = Mixed (M.Map MixedAmountKey Amount) deriving (Eq,Ord,Generic,Show)
+type MixedAmountKey = (CommoditySymbol, Maybe (CommoditySymbol, Maybe Quantity))
```

Comment:
Interesting, what's going on here ? We start to need some haddocks here.

Reply:
Yes, I can add some haddocks. These fields are 
1. The commodity symbol of the main amount,
2. `Nothing` if there is not price, or if there is a price then `Just` the following: 
      a.  The commodity symbol of the price amount
      b.  `Just` the unit price, or `Nothing` if it's a total price